StablecoinResearch/DualTokenSim/source/simulations/three_pools_simulation.py

```
@@ -0,0 +1,108 @@
+                simulation_data["collateral_free_supply_history"].append(self.collateral_token.free_supply)
+                simulation_data["virtual_pool_delta"].append(self.virtual_pool.delta)
+
+                self.market_simulator.execute_random_purchases()
+
+            if self.collateral_token.supply * self.collateral_token.price < \
+                    (self.stablecoin_token.supply * self.stablecoin_token.price) / 10e5:
+                print(
+                    "Simulation terminated early: The collateral capitalization is insufficient to support "
+                    "the stablecoin system. The algorithmic stablecoin is considered collapsed."
+                )
+                break
```

Ruff (0.12.2)

`break` outside loop

(F701)